common/src/Model/ParallelTexCoordSystem.cpp

```diff
@@ -25,13 +25,14 @@
 namespace TrenchBroom {
     namespace Model {
         ParallelTexCoordSystemSnapshot::ParallelTexCoordSystemSnapshot(ParallelTexCoordSystem* coordSystem) :
-        m_coordSystem(coordSystem),
-        m_xAxis(m_coordSystem->xAxis()),
-        m_yAxis(m_coordSystem->yAxis()) {}
+        m_xAxis(coordSystem->xAxis()),
+        m_yAxis(coordSystem->yAxis()) {}
         
-        void ParallelTexCoordSystemSnapshot::doRestore() {
-            m_coordSystem->m_xAxis = m_xAxis;
-            m_coordSystem->m_yAxis = m_yAxis;
+        void ParallelTexCoordSystemSnapshot::doRestore(TexCoordSystem* coordSystem) const {
+            ParallelTexCoordSystem *parallelCoordSystem = dynamic_cast<ParallelTexCoordSystem*>(coordSystem);
```

Comment:
Please don't use dynamic casts and use static_cast instead. Come to think of it, why do you have to cast here at all? I don't like casts, it's like tearing a hole into the type system. I try to avoid them whenever possible. In this case, you can avoid it with a double dispatch.